src/components/common/RestEntity/README.md

```diff
@@ -0,0 +1,882 @@
+const map2 = array.map(namedCallback)
+
+// anonymous, unnamed callback Component passed in
+// created and passed in at the same time
+(
+  <RestEntity.Singleton 
+    component={data => (
+      <div>
+        <h1>Program Name: {data.programname}</h1>
+        <p>Description: {data.description}</p>
+      </div>
+    )}
+  />
+)
+
+// named Component passed in as a callback here
+// it must've been created/defined elsewhere
+(
+  <RestEntity.Singleton 
+    component={ProgramCard}
+  />
+)
+```
+
+## Basic Functioning Version
+
+Let's look at our finished `ProgramComponent` now that it's using the `RestEntity`
+
+```javascript
+import React from "react";
+import RestEntity from "./RestEntity";
+
+const ProgramComponent = props => {
+  return (
+    {/* The data we care about is at the endpoint passed into `href` */}
+    <RestEntity href={`/programs/program/${props.programId}`}>
+
+      {/* This JSX will render when our data is successfully obtained */}
+      <RestEntity.Singleton 
+        component={data => (
+          <div>
+            <h1>Program Name: {data.programname}</h1>
+            <p>Description: {data.description}</p>
+          </div>
+        )}
+      />
+      
+      {/* The JSX inside this will only display if we hit an error */}
+      <RestEntity.Error>
+        <p>Oh no! There's been a problem!</p>
+      </RestEntity.Error>
+      
+      {/* The JSX inside this will only display when we're loading up data */}
+      <RestEntity.Loading>
+        <div>Loading...</div>
+      </RestEntity.Loading>
+
+    </RestEntity>
+  );
+};
+
+export default ProgramComponent;
+```
+
+Now we could have some page where we wanted to use this `ProgramComponent` and simply pass in a `programId` as `props.programId`. The component will handle the fetching of data for that program. Loading and Error messages are displayed as needed. If it successfully gets the data we want, it will render the following JSX.
+
+```javascript
+(
+<div>
+  <h1>Program Name: {data.programname}</h1>
+  <p>Description: {data.description}</p>
+</div>
+)
+```
+
+And you don't see one single of logic in that component.
+
+That's one powerful component. BUT... there's still something missing.
+
+## What's Missing? Scaling Up
+
+Our `ProgramComponent` in its current place is powerful. But is it flexible? 
+
+Not remotely. The `ProgramComponent` is going to display same JSX for any time I render it. 
+
+```javascript
+(
+<div>
+  <h1>Program Name: {data.programname}</h1>
+  <p>Description: {data.description}</p>
+</div>
+)
+```
+
+What if I wanted to add a button to that program? Or I wanted to maybe include the `programtype` or whatever. Would I have to build out a fully-featured implementation of the `RestEntity` component for one measly button?
+
+You could... but yikes. The `RestEntity` is flexible as hell. What if we tackled somme of the same inversion of control that `RestEntity` gives to us — the ability to render any sort of JSX or component.
```

Comment:
I again feel that some of these overarching statements should be moved closer to the top as well to give a more substantial introduction, and then they could be briefly reiterated again in certain sections.